recipes/libgit2/all/conanfile.py

```diff
@@ -22,6 +22,7 @@ class LibGit2Conan(ConanFile):
         "with_https": [False, "openssl", "mbedtls", "winhttp", "security"],
         "with_sha1": ["collisiondetection", "commoncrypto", "openssl", "mbedtls", "generic", "win32"],
         "with_ntlmclient": [True, False],
+        "with_regex": ["default", "builtin", "pcre", "pcre2"],
```

Comment:
What is the default? `builtin`?
Using `default` is ambiguous. Use `builtin` instead.
```suggestion
        "with_regex": ["builtin", "pcre", "pcre2"],
```

Reply:
Thank you for the review. In fact, this is how libgit2 works: https://github.com/libgit2/libgit2/blob/v1.0.1/src/CMakeLists.txt#L107-L157
```default``` doesn't have the same behavior as ```builtin```. Maybe ```default``` isn't very appropriate. What do you think about ```auto``` ?

Reply:
Thanks for the link.
If I read it correctly, then `auto` will result in a package equal to `builtin`, `pcre` or `pcre2` depending on the build machine.
I don't think `auto` will give us reproducible packages.

Maybe the safest would be to set the default to `pcre`, as we have a conan package for it.

Reply:
I see. The builtin backend is ```pcre```. But as you said, it may be preferable to set the default to conan ```pcre``` so that the dependency is explicit.
Originaly, I made ```default``` \ ```auto``` the default option so that the actual behavior isn't changed (withtout thinking about build reproductibility).

Reply:
What you can do instead, is keep the `auto` option and do something like the `cmake` recipe instead (for the ssl backend).
It has an `auto` option that aliases to `winssl` on Windows and `openssl` on others.

This requires some knowledge of you (=the recipe editor) to know what option is selected on the different platforms.
By reading the c3i build logs on clang@Linux, gcc@Linux, clang@Macos and MSVC@Windows, you can deduce what default options is selected on each platform.